### PR TITLE
feat(android): background restrictions

### PR DESCRIPTION
**Pull request by helenaford**
Video to demonstrate usage. (https://vimeo.com/459004081).

### CODE REVIEW

File: docs/react-native/docs/android/behaviour.md [hunk 1]
```diff
@@ -139,3 +139,41 @@ notifee.displayNotification({
...
+
+<Vimeo id="android-battery-optimization" caption="Android Disable Battery Optimization" />
+
+Android devices on 6.0 or higher (API level 23) there are two methods to help achieve this, `isBatteryOptimizationEnabled` and `openBatteryOptimizationSettings`.
+
+Below is a snippet on how to use these methods:
+
+```js
+// 1. checks if battery optimization is enabled
+if (notifee.isBatteryOptimizationEnabled()) {
```
Comment (mikehardy) [inline docs/react-native/docs/android/behaviour.md]:
```suggestion
const batteryOptimizationEnabled = await notifee.isBatteryOptimizationEnabled();
if (batteryOptimizationEnabled()) {
```

File: docs/react-native/docs/android/behaviour.md [hunk 2]
```diff
@@ -139,3 +139,41 @@ notifee.displayNotification({
...
+  // 2. ask your users to disable the feature
+  Alert.alert(
+      'Restrictions Detected',
+      'To ensure notifications are delivered, please disable battery optimization for the app.',
+      [
+        // 3. launch intent to navigate the user to the appropriate screen
+        {
+          text: 'OK, open settings',
+          onPress: () => notifee.openBatteryOptimizationSettings(),
```
Comment (mikehardy) [inline docs/react-native/docs/android/behaviour.md]:
```suggestion
          onPress: async () => await notifee.openBatteryOptimizationSettings(),
```

**Review by mikehardy — COMMENTED** [review]
One thing that comes up a lot on the react-native-background-geolocation-android repo about the system dialog prompts etc is some touchiness about when exactly the resolve. I'm not sure exactly what users want most (I think they want the promise to only resolve when the dialog is dismissed? Not just when it is shown? I'm not even sure what is possible - but it does seem that the first step in taking control of any of the resolution behavior is showing async/await in the example?

File: docs/react-native/docs/android/behaviour.md [hunk 3]
```diff
@@ -139,3 +139,42 @@ notifee.displayNotification({
...
+<Vimeo id="android-battery-optimization" caption="Android Disable Battery Optimization" />
+
+Android devices on 6.0 or higher (API level 23) there are two methods to help achieve this, `isBatteryOptimizationEnabled` and `openBatteryOptimizationSettings`.
+
+Below is a snippet on how to use these methods:
+
+```js
+// 1. checks if battery optimization is enabled
+const batteryOptimizationEnabled = await notifee.isBatteryOptimizationEnabled();
+if (batteryOptimizationEnabled()) {
```
Comment (helenaford) [inline docs/react-native/docs/android/behaviour.md]:
```suggestion
if (batteryOptimizationEnabled) {
```

**Comment by helenaford** [comment]
@mikehardy thanks for changes 👍 Not sure about having the promise resolve after dialog is dismissed e.t.c. Maybe it's something we can expand on.

**Comment by mikehardy** [comment]
Yeah I'm not even sure what is truly desired as mentioned but it's an area I had not even thought about before and now I'm at least listening for. Cheers :-)

**Review by mikehardy — APPROVED** [review]

**Pull request merged**